ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star-empty hide-all",
+          })
+          $star.data("story_id",res.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: res.url,
+            text: res.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${res.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          hackerNews = [];
+          //Add star if auto logged in
+          if (localStorage.getItem("Nacker")) {
+            $(".glyphicon").removeClass("hide-all");    
+          }
+        })
+      })
+    })
+  } 
+  //Function to fetch favorites
+
+  function fetchFavorites() {
+    $topiclist.empty();
+    $.ajax({
+          type: "Get",
+          url: "https://hn-favorites.herokuapp.com/stories.json",
+          headers: {
+            authorization: JSON.parse(localStorage.getItem("Nacker"))[0]
+          }
+        }).then(function(res) {
+          console.log(res);
+          hackerNews = hackerNews.concat(res);
+          hackerNews.forEach(function(value) {
+          let $li = $("<li>");
+          let $star = $ ("<span>", {
+            class: "glyphicon glyphicon-star",
+          });
+          $star.data("story_id",value.story_id);
+          $star.data("favorite_id",value.id);
+          let $text = $("<a>", {
+            class: "topic-text",
+            target: "blank",
+            href: value.url,
+            text: value.title
+          });
+          let $link = $("<span>", {
+            class: "class-filter",
+            text: `(${value.url})`
+          });
+          $li.append($star,$text,$link);
+          $topiclist.append($li);
+          });
+          hackerNews = [];
+
+        })
+  }
+
+  //Slide down forms
+  $login.on("click", function(event) {
+    $(".form-row").slideToggle(300);
+  })
+
+
+  //Top stories click
+  $topStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+  //Best stories click
+  $bestStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+  //Best stories click
+  $newStories.on("click", function(event) {
+    fetchTopNews();
+  })
+
+
+  //Toggle Icon code
+  $topiclist.on("click", ".glyphicon", function(event) {
+    console.log($(event.target));
+    if ($(event.target)[0].className === "glyphicon glyphicon-star-empty") {
+      //Post story
+      $(event.target).toggleClass("glyphicon-star-empty");
+      $(event.target).toggleClass("glyphicon-star");
+      $.get("https://hacker-news.firebaseio.com/v0/item/" + $(event.target).data().story_id + ".json?print=pretty")
```

Comment:
You're wrapping the event target with jquery multiple times - to keep from having to essentially create multiple copies of the same jquery object, you could cache the object up top. Maybe something like:

`var $tgt = $(event.target)`